syntax = "proto3";

message QuestionDetails{
  string question = 1;
  string questionClass = 2;
}

message QuestionPOSTagging{
  QuestionDetails questionDetails =1;
  string taggedQuestion = 2;
  repeated string tagSequence = 3;
}

message GetQuestionsMessage{
  int32 limit = 1;
  repeated QuestionDetails questionDetailsList=2;
  string requestId = 3;
  string fromHost = 4;
}

service QuestionClassificationServices{
  rpc getSampleQuestions(GetQuestionsMessage) returns (GetQuestionsMessage);
  rpc streamSampleQuestions(stream GetQuestionsMessage) returns (stream GetQuestionsMessage);
  rpc getTaggedQuestions(QuestionPOSTagging) returns (QuestionPOSTagging);
}